formatter/src/test_helpers.rs

```diff
@@ -14,7 +14,19 @@ macro_rules! element {
     ($($tt:tt)*) => {
         {
         let tokens = quote::quote! { $($tt)* };
-        let nodes = syn_rsx::parse2(tokens).unwrap();
+        let nodes = rstml::parse2(tokens).unwrap();
+        crate::test_helpers::get_element(nodes, 0)
+    }};
+}
+
+// Same as element, but use string representation of token stream.
+// This is usefull when testing unquoted text,
+// because current `quote!` implementation cannot provide `Span::source_text`
+// that is used in `raw_text` handler
+macro_rules! element_from_string {
+    ($val: expr) => {{
+        let tokens = <proc_macro2::TokenStream as std::str::FromStr>::from_str($val).unwrap();
```

Comment:
`indoc!` helps with correctly indenting, see the following example:

```rust
use indoc::indoc;

fn main() {
    let testing = indoc! {"
        def hello():                                <-- even though you indented your text here
            print('Hello, world!')

        hello()
    "};
    
     // notice that it starts without indentation: "def hello() ....", the indentation of the next line is only 4 spaces
    // thus it un-idents the text so that the the leftmost non-space character is in the first column
    let expected = "def hello():\n    print('Hello, world!')\n\nhello()\n";
    assert_eq!(testing, expected);
}
```

Reply:
Thanks, now I understand.